presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -0,0 +1,297 @@
+import org.testng.annotations.Test;
+
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public class TestMaterializedViewQueryOptimizer
+{
+    private static final SqlParser SQL_PARSER = new SqlParser();
+    private static final String BASE_TABLE_1 = "base_table_1";
+    private static final String BASE_TABLE_2 = "base_table_2";
+    private static final String VIEW = "view";
+
+    @Test
+    public void testWithSimpleQuery()
+    {
+        String originalViewSql = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String baseQuerySql = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String expectedRewrittenSql = format("SELECT a, b FROM %s", VIEW);
+
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+    }
+
+    @Test
+    public void testWithDistinct()
+    {
+        String originalViewSql = format("SELECT DISTINCT a, b FROM %s", BASE_TABLE_1);
+        String baseQuerySql = format("SELECT DISTINCT a, b FROM %s", BASE_TABLE_1);
+        String expectedRewrittenSql = format("SELECT DISTINCT a, b FROM %s", VIEW);
+
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        originalViewSql = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        baseQuerySql = format("SELECT DISTINCT a, b FROM %s", BASE_TABLE_1);
+        expectedRewrittenSql = format("SELECT DISTINCT a, b FROM %s", VIEW);
+
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        originalViewSql = format("SELECT DISTINCT a, b FROM %s", BASE_TABLE_1);
+        baseQuerySql = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        expectedRewrittenSql = baseQuerySql;
+
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+    }
+
+    @Test
+    public void testWithAlias()
+    {
+        String originalViewSql = format("SELECT a as mv_a, b, c as mv_c FROM %s", BASE_TABLE_1);
+        String baseQuerySql = format("SELECT a, b, c FROM %s", BASE_TABLE_1);
+        String expectedRewrittenSql = format("SELECT mv_a, b, mv_c FROM %s", VIEW);
+
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        originalViewSql = format("SELECT a as mv_a, b, c as mv_c, d FROM %s", BASE_TABLE_1);
+        baseQuerySql = format("SELECT a as result_a, b as result_b, c, d FROM %s", BASE_TABLE_1);
+        expectedRewrittenSql = format("SELECT mv_a as result_a, b as result_b, mv_c, d FROM %s", VIEW);
+
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+    }
+
+    @Test
+    public void testWithAllColumnsSelect()
+    {
+        String originalViewSql = format("SELECT * FROM %s", BASE_TABLE_1);
+        String baseQuerySql = format("SELECT * FROM %s", BASE_TABLE_1);
+        String expectedRewrittenSql = baseQuerySql;
+
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
```

Comment:
no need to reassign `baseQuerySql` to `expectedRewrittenSql`. Just directly call `assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);`. Same for all other places.